src/schemas.ts

```diff
@@ -46,6 +46,22 @@ const PackageURLWithNamespace = z
     }
   })
 
+const PackageURLString = z.string().superRefine((value, context) => {
```

Comment:
👍  Nice I really like zod's superRefine. Hopefully we can slowly tease out more of the validation logic out of the config.ts and into their own schema's here.